src/arithmetic.jl

```diff
@@ -13,37 +13,36 @@ for T in (:Taylor1, :TaylorN)
     @eval begin
         =={T<:Number,S<:Number}(a::$T{T}, b::$T{S}) = ==(promote(a,b)...)
         function =={T<:Number}(a::$T{T}, b::$T{T})
-            a, b = fixorder(a, b)
-            @simd for i in eachindex(a.coeffs)
-                @inbounds (a.coeffs[i] == b.coeffs[i]) || return false
+            if a.order == b.order
+                return all( a.coeffs .== b.coeffs )
+            elseif a.order < b.order
+                res1 = all( a.coeffs[1:a.order+1] .== b.coeffs[1:a.order+1] )
```

Comment:
For `Taylor1` this is essentially equivalent; for `TaylorN` this is a bit more involved. In any case, it's clearer the way you present it; I'll implement it.